### PR TITLE
Soapstones now have a fixed vocabulary (and rating system!)

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
add: Soapstone messages can now be rated! Attack the message with your hand to rate it positive or negative. Anyone can see the rating, and you cannot rate a message more than once, even across rounds.
del: Soapstones no longer have a write time.
tweak: Soapstones now have a fixed vocabulary to write messages with.
/:cl:

Also removed variables like `w_engraving` and such because soapstones have no writing time now.

Soapstone vocabulary is found in json and txt files in the strings folder, and you can freely edit them to change what you can write. Messages written previously will stay the same and are left as legacy.

Thank you so much to LeoZ and the rest of the folks at coderbus for sticking with me and providing help to me through several long hours of mental deterioration and frustration.

Closes #23700 - if one gets in the other one doesn't have a use really

@ChangelingRain, you requested this

### CODE REVIEW

File: code/modules/library/soapstone.dm [hunk 1]
```diff
@@ -168,6 +165,35 @@
...
+	switch(alert(user, "How would you like to rate this message?", "Message Rating", "Positive", "Negative", "Cancel"))
+		if("Positive")
+			for(var/client/C in clients)
+				if(C.ckey == creator_key)
+					C.mob << "<span class='notice'>One of your messages was rated as positive!</span>"
+					C << sound('sound/items/soapstone_rate.ogg', volume = 50)
+			user << "<span class='noticealien'>You rated this message as positive.</span>"
+			playsound(src, 'sound/items/soapstone_rate.ogg', 50, 1)
+			positive_ratings++
+			raters[user.ckey] = "positive"
```
Comment (ChangelingRain) [inline code/modules/library/soapstone.dm]: could just be TRUE/FALSE instead of a string

**Comment by ChangelingRain** [comment]
Nice

File: code/modules/library/soapstone.dm [hunk 2]
```diff
@@ -47,61 +59,46 @@
...
-	user.visible_message("<span class='notice'>[user] starts [w_engraving] a message into [T].</span>", "You start [w_engraving] a message into [T].", "<span class='italics'>You hear a [w_chipping] sound.</span>")
-	if(can_use() && do_after(user, tool_speed, target=T) && can_use())
-		if(!locate(/obj/structure/chisel_message in T))
-			user << "You [w_engrave] a message into [T]."
-			playsound(loc, 'sound/items/gavel.ogg', 50, 1, -1)
-			var/obj/structure/chisel_message/M = new(T)
-			M.register(user, message)
-			remove_use()
+	var/processed_message = replacetext(prefix, "****", suffix)
+	if(!user.Adjacent(T))
```
Comment (ChangelingRain) [inline code/modules/library/soapstone.dm]: Needs another check for if there's a message in the turf

File: strings/soapstone_suffixes.json [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+		"comfort",
+		"tears"
+	],
+	
+	"Musings": [
+		"I can't take this...",
+		"it'll happen to you too",
+		"you don't deserve this",
+		"you bastard",
+		"praise the Sun!",
```
Comment (ChangelingRain) [inline strings/soapstone_suffixes.json]: remove this and add sun as an object

File: strings/soapstone_suffixes.json [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+		"despair",
+		"light",
+		"dark",
+		"bravery",
+		"resignation",
+		"comfort",
+		"tears"
+	],
+	
+	"Musings": [
```
Comment (ChangelingRain) [inline strings/soapstone_suffixes.json]: i'm not sure how well this works as a suffix part, as you can't add it in addition to other stuff so it's just "be wary of I can't take this..." or "praise the praise the sun"

Comment (Xhuis) [inline strings/soapstone_suffixes.json]: That's intentional. If people want to write things that make no sense, all the power to them.

**Comment by ChangelingRain** [comment]
This should probably be tgui but I'll do that later once this is merged

**Comment by Armhulen** [comment]
new sprites :+1: it looks a lot more in style with the game 

fixed vocab :-1: i've had so much fun reading these and i feel like the station has culture for the server you're on, PLEASE don't do this

either way :-1: atomize your prs

**Comment by Haevacht** [comment]
Your writing message sound is the phantom spawnin sound.

If you are gonna rip sounds straight from the game DO IT RIGHT.
But actually don't.

File: strings/soapstone_suffixes.json [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+		"handcuffs",
+		"plant",
+		"food",
+		"drink",
+		"tool",
+		"amazing item",
+		"amazing loot",
+		"amazing clothing",
+		"amazing drugs",
+		"amazing rack"
```
Comment (Iamgoofball) [inline strings/soapstone_suffixes.json]: excuse me why isn't Sun in here

**Review by Iamgoofball — CHANGES_REQUESTED** [review]

File: strings/soapstone_suffixes.json (hunk 5 above)
Comment (Haevacht) [inline strings/soapstone_suffixes.json]: Or amazing chest?

**Comment by jaxy15** [comment]
White soapstone for co-op mining when?

**Comment by Xhuis** [comment]
@Haevacht The actual sound for the orange soapstone was so bad quality that I couldn't use it.
@Armhulen While I'd like to say that I support a free-type message system, it doesn't work. Joan requested this because people would just write random bullshit and copy-paste huge chunks of text, and having a fixed vocabulary both inspires people to be creative with it ("lizard ahead" in front of a monkey pen, or "try walking" in the janitor's office) and prevent metagame and abuse.

File: strings/soapstone_suffixes.json (hunk 5 above)
Comment (Xhuis) [inline strings/soapstone_suffixes.json]: "praise the Sun!" is in musings.

**Comment by Xhuis** [comment]
Since @Armhulen raised a kerfuffle about the sprites, I cut them out and will be pushing them in a different PR. Sounds won't be in that PR since code is changed and would cause conflicts, but will be coming after.

I figured that using the actual sounds wouldn't be an issue because of all the ripped sounds we have already. If it is then please tell me!

**Comment by optimumtact** [comment]
you should go back to compiling

**Comment by Haevacht** [comment]
Doesn't change the fact the sounds are still straight ripped from the game, which is bad.

Silent messages would work anyway.

**Comment by Xhuis** [comment]
@optimumtact You can't control me
@Haevacht It doesn't matter here, since they aren't even used :P

**Comment by Haevacht** [comment]
There's been like 7 soapstone PRs I am allowed to get them confused.

**Comment by Xhuis** [comment]
No you aren't. This is your life now.

**Comment by jughu** [comment]
@xhuis remove the crate from cargo REEEE i scribbled loke shitcurity in every cell infront of brig and doors

**Comment by Xhuis** [comment]
what.

**Comment by jughu** [comment]
Its not like anyone can order engraved scribble crates from cargo and infinitly carve shit everywhere ya know... i could atleast order 10-15 librarian crates a round ans thats 45 carving uses

**Comment by Xhuis** [comment]
If people want to do that's let 'em. I don't want to include balance changes in this PR.

**Comment by coiax** [comment]
@jughu spamming stupid messages is exactly the sort of shit that makes the feature controversial, stop it.

File: code/modules/library/soapstone.dm [hunk 6]
```diff
@@ -1,22 +1,40 @@
+#define SOAPSTONE_PREFIX_FILE "strings/soapstone_prefixes.txt"
+#define SOAPSTONE_SUFFIX_FILE "soapstone_suffixes.json"
+//Vocabulary lists; soapstones use a prefix and a suffix. Optionally, they can have a prefix and suffix, then a conjunction that links another set.
+var/global/list/soapstone_prefixes = list() //Read from "strings/soapstone_prefixes.txt"; if you're adding your own, put **** where the subject should be!
+var/global/list/soapstone_suffixes = list() //Read from "strings/soapstone_suffixes.json"
 /obj/item/soapstone
 	name = "chisel"
-	desc = "Leave informative messages for the crew, including the crew of future shifts!\nEven if out of uses, it can still be used to remove messages.\n(Not suitable for engraving on shuttles, off station or on cats. Side effects may include beatings, bannings and orbital bombardment.)"
+	desc = "Leave \"informative\" messages for the crew, including the crew of future shifts!\n\
```
Comment (coiax) [inline code/modules/library/soapstone.dm]: BLAH BLAH NAZIS HITLER, LINE BREAKS

Comment (ChangelingRain) [inline code/modules/library/soapstone.dm]: Fuck off you can see the \n right there

Comment (Xhuis) [inline code/modules/library/soapstone.dm]: REEEEEEEEEEEEEEE

Comment (coiax) [inline code/modules/library/soapstone.dm]: Yes, I know, I wrote it.

**Comment by coiax** [comment]
I would like to see if the spamming and abuse problem is fixed by free removals. I mean, the admins really should be dealing with that, but apparently not.

I've been using messages as a form of developer commentary, tagging places with notes about how the map has changed, and that's obviously not possible with a limited vocab.

**Comment by Xhuis** [comment]
You can freely varedit the contents to anything. Vocabulary is only used when manually writing.

**Comment by coiax** [comment]
I'm not an admin, Xhuis, I can't varedit things.

**Comment by Xhuis** [comment]
Oh :(

**Comment by KorPhaeron** [comment]
Man I didn't even see this was open

Entirely uneeded now that soapstones could remove for free, there was no longer any spam

**Comment by KorPhaeron** [comment]
This completely kills all the fun of them, what a shame. Why even have persistent messages for the sake of "praise the lizard" written in random spots?

@coiax You should revert this at some point, I liked seeing messages responding to each other across rounds and telling little stories.

**Comment by Xhuis** [comment]
In Dark Souls, where the fixed vocabulary also existed, it was fun seeing how people found creative ways to write warnings, jokes, etc. with a fixed vocabulary. It also gave an effective way of preventing abuse; spamming "SHITCURITY" over every brig tile could only be fixed if most of the messages showed up in one round and someone actually had a soapstone. Generally, random messages like the one you mentioned are ignored and don't get any ratings, but actually clever stuff gets hundreds of appraisals.

**Comment by coiax** [comment]
I'll see how the feature goes with a restricted vocabulary. I will obviously keep the rating system, although I dunno why we couldn't just use my UI with the admin delete buttons (and ghost rating).

**Comment by coiax** [comment]
@KorPhaeron a revert would mean more coming from the non-author, as I don't want to get a reputation for code babying as it is. Anyway, I respect @Xhuis, so I'll see how this pans out.

**Comment by Xhuis** [comment]
Thank you very much, coiax. As for the UI and ghost rating, hell, go for it. It just got closed when the PR was merged, I don't see why it can't be added.

**Comment by optimumtact** [comment]
it's fine to revert it then, I don't really midn either way I just assuemd this was the modern version fo the other pr because it did have the extra features

**Pull request merged**